src/bugsplat/bugsplat.py

```diff
@@ -1,100 +1,120 @@
 import json
-import os
-import requests
 import traceback
-import sys
+from logging import Logger, getLogger
+from os import PathLike
+from pathlib import PurePath
+from typing import List
+
+import requests
+
 
 class BugSplat:
-    def __init__(self, database, application, version):
+    def __init__(self,
+                 database: str,
+                 application: str,
+                 version: str,
+                 logger: Logger = None):
         self.database = database
         self.application = application
         self.version = version
-        self.additionalFilePaths = []
-        self.appKey = ''
+        self.additional_file_paths = []
+        self.app_key = ''
         self.description = ''
         self.email = ''
         self.user = ''
+        self.logger = logger or getLogger(__name__)
 
-    def setDefaultAdditionalFilePaths(self, additionalFilePaths):
-        self.additionalFilePaths = additionalFilePaths
+    def set_default_additional_file_paths(self, additional_file_paths: List[PathLike]):
+        self.additional_file_paths = additional_file_paths
 
-    def setDefaultAppKey(self, key):
-        self.appKey = key
+    def set_default_app_key(self, key: str):
+        self.app_key = key
 
-    def setDefaultDescription(self, description):
+    def set_default_description(self, description: str):
         self.description = description
-            
-    def setDefaultEmail(self, email):
+
+    def set_default_email(self, email: str):
         self.email = email
 
-    def setDefaultUser(self, user):
+    def set_default_user(self, user: str):
         self.user = user
 
-    def post(self, ex, additionalFilePaths = [], appKey = '', description = '', email = '', user = ''):
-        print('\n')
-        print('BugSplat caught an Unhandled Exception!')
-        print('\n')
+    # TODO Remove these completely
```

Comment:
happy to just do a major version rev

Reply:
that would be great! There are already technically other breaking changes but those are the biggest.